### ISSUE TITLE
How to add Custom names?

### ISSUE THREAD

**Issue by sunilpro** [facebookresearch/wav2letter#480]
How to do inference/decode on audio/speech containing custom names of people or cities, etc?
Is this already supported by wav2letter? Do I need to retrain the models with audio containing the names?

**Comment by vineelpratap** [comment]
Hi, how large is your training data in hours. It is recommended that you retrain the model using your audio data adapting this recipe - https://git.io/JvIjJ. 

You might have to adjust the dropout in the architecture for best results.

**Comment by avidov** [comment]
You can also try to simply add the custom names to the lexicon.

**Comment by sunilpro** [comment]
@vineelpratap I am not retraining the acoustic model. I just tried adding my name "Sunil" into the lexicon.txt and also added few sentences with Sunil and retrained only the kenlm language model.

But its not getting recognized, I don't even see the phonetic equivalant of the recognition either.

Is it possible to see the acoustic model's output to know/debug what acoustic model's output is before it goes to CTC or LM?

**Comment by vineelpratap** [comment]
@sunilpro - It is not generated because LM doesn't contain it and thus assigns a very low score for the word.  

> Is it possible to see the acoustic model's output to know/debug what acoustic model's output is before it goes to CTC or LM?

Yes, you can do using token level decoding. @xuqiantong should be able to guide you with the process for online decoder.

**Comment by xuqiantong** [comment]
@sunilpro Yea, you need to make sure your kenlm contains `Sunil` and you provided a reasonable spelling of it in the lexicon. 

> Is it possible to see the acoustic model's output to know/debug what acoustic model's output is before it goes to CTC or LM?
You can simply set the beamsize to 1 and all the scores to 0 to check the greedy path generated by the AM.

**Comment by tlikhomanenko** [comment]
Closing because of no activity, feel free to reopen!
